rwlib/source/rw/bit_cast.cpp

```diff
@@ -0,0 +1,19 @@
+#ifndef _LIBRW_BIT_CAST_CPP_
+#define _LIBRW_BIT_CAST_CPP_
+
+//Based on https://gist.github.com/socantre/3472964
+#include <cstring> // memcpy
+#include <type_traits> // is_trivially_copyable
+#include "rw/defines.hpp" // RW_ASSERT
+
+template <class Dest, class Source>
+inline Dest bit_cast(Source const &source) {
+    RW_ASSERT(std::is_trivially_copyable<Dest>::value);
+    RW_ASSERT(std::is_trivially_copyable<Source>::value);
+
+    Dest dest = Dest{};
+    std::memcpy(&dest, &source, sizeof(Dest));
```

Comment:
You don't check that sizeof(Dest) <= sizeof(Source).

Is it necessary to specify the Source where bit_cast is used? It should be deduced.